### PR TITLE
set xor min trie thingy

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/ys-SetXORMin.mdx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <set>
+using namespace std;
+
+struct trie {
```
Comment (SansPapyrus683) [inline solutions/ys-SetXORMin.mdx:26]: rename to `Trie`

File: solutions/ys-SetXORMin.mdx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	trie* children[2] = {};
+	int cnt = 0;
```
Comment (SansPapyrus683) [inline solutions/ys-SetXORMin.mdx:28]: `cnt`? `cnt` what?

File: solutions/ys-SetXORMin.mdx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+void add(trie* root, int x) {
```
Comment (SansPapyrus683) [inline solutions/ys-SetXORMin.mdx:31]: ever heard of property methods?

File: solutions/ys-SetXORMin.mdx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	while (q--) {
+		int t, x; 
+		cin >> t >> x;
+		if (t == 0) {
+			if (!s.count(x)) {
+				s.insert(x);
+				add(&tr, x);
+			}
+		}
+		else if (t == 1) {
```
Comment (SansPapyrus683) [inline solutions/ys-SetXORMin.mdx:90]:
```suggestion
		} else if (t == 1) {
```

File: solutions/ys-SetXORMin.mdx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+			if (s.count(x)) {
+				s.erase(x);
+				remove(&tr, x);
+			}
+		}
+		else {
```
Comment (SansPapyrus683) [inline solutions/ys-SetXORMin.mdx:96]:
```suggestion
		} else {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/ys-SetXORMin.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/ys-SetXORMin.mdx:31]: erm doesnt rly matter

Comment (SansPapyrus683) [reply solutions/ys-SetXORMin.mdx]: it's good practice

Comment (envyaims) [reply solutions/ys-SetXORMin.mdx]: ok but like it really doesnt matter in this simple case

Comment (envyaims) [reply solutions/ys-SetXORMin.mdx]: i dont think itll work out in this case lol

Comment (SansPapyrus683) [reply solutions/ys-SetXORMin.mdx]: why not

Comment (envyaims) [reply solutions/ys-SetXORMin.mdx]: nastier implem

File: solutions/ys-SetXORMin.mdx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	return res;
+}
+
+int main() {
+	cin.tie(0) -> sync_with_stdio(0);
+	Trie tr;
+	int q; 
+	cin >> q;
+	set<int> s;
+	while (q--) {
```
Comment (Aadit-Ambadkar) [inline solutions/ys-SetXORMin.mdx:81]: you're gonna get roasted about this, so might as well change it.
```suggestion
	for (int qry = 0; qry < q; qry++) {
```

**Review by Aadit-Ambadkar — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Comment by Aadit-Ambadkar** [comment]
@SansPapyrus683 2 approvals

File: solutions/ys-SetXORMin.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/ys-SetXORMin.mdx]: @Aadit-Ambadkar until this is addressed i won't merge

Comment (envyaims) [reply solutions/ys-SetXORMin.mdx]: @SansPapyrus683 im not really sure what you mean. I think this version is fine but if you want to edit it with your suggestion feel free.

Comment (SansPapyrus683) [reply solutions/ys-SetXORMin.mdx]: i mean do you know have a method belong to a class?

Comment (envyaims) [reply solutions/ys-SetXORMin.mdx]: yeah, but im traversing on the object itself. I think i'd have to make the function recursive for that to work which might make it harder to read :/

**Pull request merged**